### PR TITLE
Git repository overwrite and sql migration

### LINKED ISSUE #3386: Project and Service scoped Git repository
### Feature description

Currently, the Git repository is connected directly to the service; as we move forward in our plans to make a project hierarchy, we want to have a project scoped git connection.
A project scoped git repository will be the automatic default for all the services inside the project.
The user can overwrite the project repository and set a custom service repository as an option.
I attached a screenshot from our Figma to show the UI for this situation.

The main task compose from two smaller ones, the UI and the backend
- [ ] Custom git repositories for services (UI)
- [ ] Custom git repositories for services (Backend)

Service - GitHub sync b

### Use case

_No response_

### Are you willing to submit PR?

_No response_

### PR DESCRIPTION
**Pull request by tupe12334**
closes #3386 

## PR Details

This pr changes the data models from a Git Repository that saves the connection to Resource to Resource that saves the connection to Git Repository

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/git/git.resolver.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class GitResolver {
   )
   async deleteGitRepository(
     @Args() args: DeleteGitRepositoryArgs
-  ): Promise<Resource> {
+  ): Promise<boolean> {
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/git/git.resolver.ts:71]: why did you decide to change the return type?

File: packages/amplication-server/src/core/git/git.resolver.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class GitResolver {
   async createGitRepository(
     @Args() args: CreateGitRepositoryArgs
   ): Promise<Resource> {
-    return this.gitService.createGitRepository(args.data);
+    return this.gitService.createRemoteGitRepository(args.data);
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/git/git.resolver.ts:36]: why did you decide to change the name?

File: packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts [hunk 3]
```diff
@@ -35,4 +35,7 @@ export class ResourceWhereInput {
     description: undefined
   })
   project?: WhereUniqueInput | null;
+
+  @Field(() => String, { nullable: true })
+  projectId?: string | null;
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts:40]: why did you add `projectId` while we already have the property `project`

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 4]
```diff
@@ -109,10 +110,23 @@ export class ResourceService {
       throw new InvalidColorError(color);
     }
 
+    if (args.data.resourceType === EnumResourceType.ProjectConfiguration) {
+      throw new AmplicationError(
+        'Try to create project configuration from createResource function'
+      );
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.ts:116]: Resource of type ProjectConnfiguration cannot be created manually

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 5]
```diff
@@ -109,10 +110,23 @@ export class ResourceService {
...
+    const projectId = args.data.project.connect.id;
+
+    const projectConfiguration = await this.projectConfiguration(projectId);
+
     const resource = await this.prisma.resource.create({
       data: {
         ...DEFAULT_SERVICE_DATA,
         ...args.data,
+        gitRepository: {
+          connect: { id: projectConfiguration.gitRepositoryId }
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.ts:128]: This will throw an exception in case projectConfiguration is null (which should not happen logically, but still)

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 6]
```diff
@@ -337,20 +351,6 @@ export class ResourceService {
       throw new Error(INVALID_DELETE_PROJECT_CONFIGURATION);
     }
 
-    const gitRepo = await this.prisma.gitRepository.findUnique({
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.ts:340]: you can still delete the gitRepo record in case `override` is true - knowing that the resource we delete is not the  projectConfiguration

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 7]
```diff
@@ -609,17 +609,37 @@ export class ResourceService {
...
+      })
+    ).gitRepository.gitOrganization;
   }
 
   async project(resourceId: string): Promise<Project> {
-    return this.prisma.project.findUnique({ where: { id: resourceId } });
+    return this.prisma.resource
+      .findUnique({ where: { id: resourceId } })
+      .project();
+  }
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/resource/resource.service.ts:635]: we agreed that we should remove this function and the resolver should use the project service instead of the resource service

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
please see my comments

File: packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts (hunk 3 above)
Comment (tupe12334) [reply packages/amplication-server/src/core/resource/dto/ResourceWhereInput.ts:40]: To match more closely to the Prisma orm

File: packages/amplication-server/src/core/git/git.resolver.ts (hunk 2 above)
Comment (tupe12334) [reply packages/amplication-server/src/core/git/git.resolver.ts:36]: To match our convention of calling remote for the non DB git repositories

File: packages/amplication-server/src/core/git/git.resolver.ts (hunk 1 above)
Comment (tupe12334) [reply packages/amplication-server/src/core/git/git.resolver.ts:71]: From now on, which Resource to return?
Git repository have several resources.
So I checked the usage of this endpoint and tested the I don't change existing behavior.
